contrib/babelfishpg_tds/src/backend/tds/tdsresponse.c

```diff
@@ -1719,7 +1720,16 @@ PrepareRowDescription(TupleDesc typeinfo, List *targetlist, int16 *formats,
 
 				/* fetch the relation name, schema name */
 				relMetaDataInfo->partName[0] = RelationGetRelationName(rel);
-				relMetaDataInfo->partName[1] = get_namespace_name(RelationGetNamespace(rel));
+				physical_schema_name = get_namespace_name(RelationGetNamespace(rel));
+
+				/* logical schema name should be sent as part of response as far as Babelfish is concerned. */
+				if (pltsql_plugin_handler_ptr && 
+					pltsql_plugin_handler_ptr->pltsql_get_logical_schema_name)
+					/* Its very unlikely that logical schema name could not be found for any other cases. */
```

Comment:
is it possible that we do not find the logical schema name? what would be the behavior in that case?

Reply:
We should always get logical schema name given physical schema because TDS receives information of physical schema from engine. Raising an error would be the best choice if we do not get logical schema for some physical schema which we are already doing.